packages/mds-agency/handlers/create-event.ts

```diff
@@ -138,20 +138,28 @@ export const createEventHandler = async (
     await refreshDeviceCache(device, event)
 
     const { telemetry } = event
+    let recorded_telemetry: Recorded<Telemetry> | undefined = undefined
     if (telemetry) {
-      await db.writeTelemetry(normalizeToArray(telemetry))
+      const results = await db.writeTelemetry(normalizeToArray(telemetry))
+      if (results.length > 0) {
+        recorded_telemetry = results[0]
+      }
     }
 
     // database write is crucial; failures of cache/stream should be noted and repaired
     const recorded_event = await db.writeEvent(event)
 
     try {
-      await Promise.all([
+      const promises = [
         cache.writeEvents([recorded_event]),
         stream.writeEvent(recorded_event),
-        cache.writeTelemetry([telemetry]),
-        stream.writeTelemetry([telemetry])
-      ])
+        cache.writeTelemetry([telemetry])
+      ]
+      if (recorded_telemetry) {
+        promises.push(stream.writeTelemetry([recorded_telemetry]))
+      }
+
+      await Promise.all(promises)
```

Comment:
We have a pattern for conditional promises we use all over, that would have this looking something like:
```typescript
await Promise.all([cache.writeEvents([recorded_event]), stream.writeEvent(recorded_event), ...(recorded_telemetry ? [cache.writeTelemetry([recorded_telemetry]), stream.writeTelemetry([recorded_telemetry])] : []))
```